earthfile2llb/wait_block.go

```diff
@@ -0,0 +1,194 @@
+package earthfile2llb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/earthly/earthly/states"
+	"github.com/earthly/earthly/util/llbutil"
+	"github.com/earthly/earthly/util/llbutil/pllb"
+	"github.com/moby/buildkit/exporter/containerimage/exptypes"
+	"github.com/moby/buildkit/frontend/gateway/client"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/pkg/errors"
+)
+
+type saveImagesWaitItem struct {
+	converters []*Converter
+	images     []states.SaveImage
+	push       []bool
+}
+
+type runCommandWaitItem struct {
+	c   *Converter
+	cmd *pllb.State
+}
+
+type waitItem interface {
+	wait(context.Context) error
+}
```

Comment:
If I understand correctly only `RUN` and `SAVE IMAGE` commands cause waiting within the wait block. I wonder if `FROM` or `GIT CLONE` or other commands with external interaction could benefit from waiting too in some cases...

Reply:
We should wait for these in a case such as

```
WAIT
    GIT CLONE ...
END
```

but for a case such as:

```
foo:
    FROM alpine
    GIT CLONE ...

WAIT
    BUILD +foo
END
```

I need to add in a wait under the finalizeState (of `+foo`), to add in the last command; however i'm currently fighting with a 

```
time="2022-05-17T00:15:47Z" level=error msg="/moby.buildkit.v1.Control/Solve returned error: rpc error: code = Canceled desc = context canceled\n"
```

error which is preventing me from pushing my latest attempt to implement this.